File: Assets/Bundles/Server/Resources/TS/Signals/EntityDamageServerSignal.lua
-- Compiled with unity-ts v2.1.0-75
local Cancellable = require("Shared/TS/Util/Cancellable").Cancellable
local EntityDamageServerSignal
do
	local super = Cancellable
	EntityDamageServerSignal = setmetatable({}, {
		__tostring = function()
			return "EntityDamageServerSignal"
		end,
		__index = super,
	})
	EntityDamageServerSignal.__index = EntityDamageServerSignal
	function EntityDamageServerSignal.new(...)
		local self = setmetatable({}, EntityDamageServerSignal)
		return self:constructor(...) or self
	end
	function EntityDamageServerSignal:constructor(entity, amount, damageType, fromEntity)
		super.constructor(self)
		self.entity = entity
		self.amount = amount
		self.damageType = damageType
		self.fromEntity = fromEntity
	end
end
return {
	EntityDamageServerSignal = EntityDamageServerSignal,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Bundles/Imports/Core/Core~/src/Server/Signals/EntityDamageServerSignal.ts
import { DamageType } from "Shared/Damage/DamageType";
import { Entity } from "Shared/Entity/Entity";
import { Cancellable } from "Shared/Util/Cancellable";

export class EntityDamageServerSignal extends Cancellable {
	constructor(
		public readonly entity: Entity,
		public amount: number,
		public damageType: DamageType,
		public fromEntity?: Entity,
		public canDamageAllies?: boolean,
	) {
		super();
	}
}
